perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java

```diff
@@ -51,11 +55,294 @@
 	}
 
+	@Override
+	public Consent createConsent(PerunSession perunSession, Consent consent) throws ConsentNotExistsException, ConsentExistsException, ConsentHubNotExistsException {
+		// Check if consent already exists
+		if(consentExists(perunSession, consent)){
+			throw new ConsentExistsException("Consent already exists.");
+		}
+
+		// Remove UNSIGNED consents from consent hub
+
+		try {
+			Consent consentExisting = getConsentForUserAndConsentHub(perunSession, consent.getUserId(), consent.getConsentHub().getId(), ConsentStatus.UNSIGNED);
+			if (consentExisting != null) {
+				deleteConsent(perunSession, consentExisting);
+			}
+		} catch (ConsentNotExistsException e) {
+			// IGNORE
+		}
+
+		try {
+			int newId = Utils.getNewId(jdbc, "consents_id_seq");
+
+			jdbc.update("insert into consents(id, user_id, consent_hub_id, status, created_at, created_by, modified_at, modified_by, created_by_uid, modified_by_uid) " +
+					"values (?,?,?,?::consent_status,"+Compatibility.getSysdate()+",?,"+Compatibility.getSysdate()+",?,?,?)", newId, consent.getUserId(),
+				consent.getConsentHub().getId(), ConsentStatus.UNSIGNED.toString(), perunSession.getPerunPrincipal().getActor(), perunSession.getPerunPrincipal().getActor(),
+				perunSession.getPerunPrincipal().getUserId(), perunSession.getPerunPrincipal().getUserId());
+			log.info("Consent {} created.", consent);
+			consent.setId(newId);
+
+			//TODO CREATE RELATIONSHIP IN DATABASE for attributes
+
+			return consent;
+		} catch (Exception e){
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void deleteConsent(PerunSession perunSession, Consent consent) throws ConsentNotExistsException {
+		// Check if consent exists
+		checkConsentExists(perunSession, consent);
+
+		try {
+			int numberOfRows = jdbc.update("delete from consents where id=?", consent.getId());
+			if (numberOfRows != 1) throw new ConsentNotExistsException(consent);
+			log.info("Consent {} deleted.", consent);
+			consent.setAttributes(null);
+			removeAttrsForConsent(perunSession, consent.getId());
+		} catch (Exception e){
+			throw new InternalErrorException(e);
+		}
+	}
+
+
+	@Override
+	public List<Consent> getAllConsents(PerunSession sess) {
+		List<Consent> result = new ArrayList<>();
+		List<ConsentHub> consentHubs = getAllConsentHubs(sess);
+		for (ConsentHub consentHub : consentHubs) {
+			result.addAll(getConsentsForConsentHub(sess, consentHub.getId()));
+		}
+		return result;
+	}
+
+
+	@Override
+	public List<Consent> getConsentsForConsentHub(PerunSession sess, int id, ConsentStatus status) {
+		try {
+			List<Consent> consents = jdbc.query("select " + consentMappingSelectQuery + " from consents where consents.consent_hub_id=? and consents.status=?::consent_status ", CONSENT_MAPPER_WITHOUT_HUB, id, status.toString());
+			ConsentHub consentHub = getConsentHubById(sess, id);
+			consentHub.setFacilities(getFacilitiesForConsentHub(consentHub));
```

Comment:
I think this is already part of the `getConsentHubById` method.

Reply:
You're right.

Reply:
You're right.